samza-test/src/test/java/org/apache/samza/test/table/TestCouchbaseRemoteTableEndToEnd.java

```diff
@@ -0,0 +1,165 @@
+    BucketConfiguration config = new BucketConfiguration();
+    config.numNodes = 1;
+    config.numReplicas = 1;
+    config.name = bucketName;
+    return config;
+  }
+
+  protected void initClient() {
+    couchbaseEnvironment = DefaultCouchbaseEnvironment.builder()
+        .bootstrapCarrierDirectPort(couchbaseMock.getCarrierPort("inputBucket"))
+        .bootstrapHttpDirectPort(couchbaseMock.getHttpPort())
+        .build();
+    cluster = CouchbaseCluster.create(couchbaseEnvironment, "couchbase://127.0.0.1");
+  }
+
+  protected void shutdownMock() {
+    cluster.disconnect();
+    couchbaseEnvironment.shutdownAsync().toBlocking().single();
+    couchbaseMock.stop();
+  }
+
+  @Test
+  public void testDummy() throws Exception {
+    List<String> bucketNames = new ArrayList<>();
+    String inputBucketName = "inputBucket";
+    String outputBucketName = "outputBucket";
+    bucketNames.add(inputBucketName);
+    createMockBuckets(bucketNames);
+    initClient();
+
+    Bucket inputBucket = cluster.openBucket(inputBucketName);
+
+    inputBucket.upsert(ByteArrayDocument.create("1", "a".getBytes()));
+    inputBucket.upsert(ByteArrayDocument.create("2", "b".getBytes()));
+    inputBucket.upsert(ByteArrayDocument.create("3", "c".getBytes()));
+    inputBucket.upsert(ByteArrayDocument.create("4", "d".getBytes()));
+
+//    int count = 10;
+    String[] pageViews = new String[]{"1", "2", "3", "4"};
+
+    int partitionCount = 1;
+    Map<String, String> configs = TestLocalTableEndToEnd.getBaseJobConfig(bootstrapUrl(), zkConnect());
+
+    configs.put("streams.PageView.samza.system", "test");
+    configs.put("streams.PageView.source", Base64Serializer.serialize(pageViews));
+    configs.put("streams.PageView.partitionCount", String.valueOf(partitionCount));
+    Config config = new MapConfig(configs);
+
+    final StreamApplication app = appDesc -> {
+      DelegatingSystemDescriptor inputSystemDescriptor = new DelegatingSystemDescriptor("test");
+      GenericInputDescriptor<String> inputDescriptor =
+          inputSystemDescriptor.getInputDescriptor("PageView", new NoOpSerde<>());
+
+      CouchbaseTableReadFunction<String> readFunction =
+          new CouchbaseTableReadFunction<>(String.class)
+          .withBootstrapCarrierDirectPort(couchbaseMock.getCarrierPort(inputBucketName))
+          .withBootstrapHttpDirectPort(couchbaseMock.getHttpPort())
+          .withClusters(Collections.singletonList("couchbase://127.0.0.1"))
+          .withBucketName(inputBucketName)
+          .withSerde(new StringSerde());
+
+      System.out.println(readFunction.toString());
+
+      RemoteTableDescriptor<String, String> inputTableDesc = new RemoteTableDescriptor<>("input-table");
+      inputTableDesc.withReadFunction(readFunction);
+      Table<KV<String, String>> inputTable = appDesc.getTable(inputTableDesc);
+
+      appDesc.getInputStream(inputDescriptor).map(k -> KV.of(k, k)).join(inputTable, new JoinFunction()).map(n -> {
```

Comment:
Please add assertions to make sure it's working.

Reply:
Sure. I am changing System.out.println to assertEquals